### PR TITLE
CLI Display Mirror with Braille Rendering

### PR DESCRIPTION
**Pull request by FinallyEve**
## Summary

Adds a high-fidelity display mirror to the CLI simulator using Unicode braille characters (U+2800-U+28FF), providing 4x the detail of half-block characters in the same terminal footprint. The mirror renders each device's 128x64 OLED display alongside readable text captions in a hybrid layout.

### Features

- **Braille rendering** — `renderToBraille()` maps 2x4 pixel grids to braille codepoints, producing 64x16 character output from a 128x64 pixel buffer
- **XBM bitmap decoding** — `decodeXBMToBuffer()` and `drawImage()` properly decode XBM format (LSB-first, row-major) into the pixel buffer
- **Hybrid display layout** — When mirror is ON, captions float alongside the first 4 lines of ASCII art; when OFF, captions display alone
- **`mirror` command** — Toggle, on, off (aliases: `m`)
- **`captions` command** — Toggle, on, off (aliases: `cap`, `cc`)
- **`display` command** — Toggles both mirror and captions together; forces both ON if they disagree (aliases: `d`)
- **`reboot` command** — Resets device state machine back to initial state (aliases: `restart`)

### Test Coverage

41 new tests (80 total), covering:
- Display driver text history, XBM decoding, image compositing, ASCII rendering
- Braille rendering (dimensions, blank, content, full block, dot mapping)
- All CLI commands (mirror, captions, display, reboot) with toggle/on/off/alias/edge cases

## Test plan

- [x] `pio test -e native_cli_test` — all 80 tests pass
- [ ] `pio run -e native_cli` — build simulator, run with `mirror on`, verify braille art + captions side-by-side
- [ ] Verify `display on`/`display off` toggles both mirror and captions
- [ ] Verify `captions off` hides captions while mirror remains visible

### CODE REVIEW

File: include/device/drivers/native/native-display-driver.hpp [hunk 1]
```diff
@@ -270,6 +268,46 @@ class NativeDisplayDriver : public DisplayDriverInterface {
...
+     * vs 2 bits for half-block characters. Same 64x16 terminal footprint
+     * but dramatically better text legibility.
+     *
+     * Braille dot layout:
+     *   (0,0)=0x01  (1,0)=0x08
+     *   (0,1)=0x02  (1,1)=0x10
+     *   (0,2)=0x04  (1,2)=0x20
+     *   (0,3)=0x40  (1,3)=0x80
+     */
+    std::vector<std::string> renderToBraille() const {
```
Comment (FinallyEve) [inline include/device/drivers/native/native-display-driver.hpp]: results kinda speak for themselves:

Image

**Review by FinallyEve — COMMENTED** [review]
Pretty cool to pull the repo and push a legit feature in an evening :) Agentic paired programming for the win ✊
